File: 345_Reverse_Vowels.js
var reverseVowels = function (s) {
    vowels = {"a":true, "o":true, "e":true, "i":true, "u":true, "A":true, "O":true, "E":true, "I":true, "U":true};
    i = 0;
    j = s.length - 1;
    l = s.split();//字符串转成数组
    while (i < j) {
        if (l[i] in vowels && l[j] in vowels) {
            i++;
            j--;
        }
        if (!(l[i] in vowels)) {
            i++;
        }
        if (!(l[j] in vowels)) {
            j--;
        }
        t = l[i];
        l[i] = l [j];
        l[j] = t;
    }  
    return l.join("");//数组转成字符串
};

File: 345_Reverse_Vowels.py
def reverseVowels(s):
    """
    :type s: str
    :rtype: str
    """
    vowels = ["a", "o", "e", "i", "u", "A", "O", "E", "I", "U"]
    i = 0
    j = len(s)
    l = list(s)//字符串转成列表
    while  i < j-1:
        if l[i] in vowels and l[j - 1] in vowels:
            l[i], l[j - 1] = l[j - 1], l[i]
            i += 1
            j -= 1
        if l[i] not in vowels:
            i += 1
        if l[j-1] not in vowels:
            j -= 1
    return "".join(l)